addons/html_builder/static/src/website_builder/plugins/menu_data_plugin.js

```diff
@@ -0,0 +1,69 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { NavbarLinkPopover } from "@html_editor/main/link/navbar_link_popover";
+import { MenuDialog, EditMenuDialog } from "@website/components/dialog/edit_menu";
+import { user } from "@web/core/user";
+import { withSequence } from "@html_editor/utils/resource";
+
+class MenuDataPlugin extends Plugin {
+    static id = "menuDataPlugin";
+    static dependencies = ["overlay"];
+    resources = {
+        link_popovers: [
+            withSequence(10, {
+                Popover: this.dependencies.overlay.createOverlay(
+                    NavbarLinkPopover,
+                    {
+                        closeOnPointerdown: false,
+                    },
+                    { sequence: 50 }
+                ),
+                isAvailable: (linkElement) =>
+                    linkElement &&
+                    linkElement.closest(".top_menu, o_extra_menu_items, [data-content_menu_id]") &&
+                    !linkElement.closest(
+                        ".dropdown-toggle, li.o_header_menu_button a, [data-toggle], .o_offcanvas_logo, .o_mega_menu"
+                    ),
+                props: (props) => ({
+                    ...props,
+                    checkIsWebsiteDesigner: () => user.hasGroup("website.group_website_designer"),
+                    onClickEditLink: (elem) => {
+                        const menuEl = elem.props.linkElement.querySelector("[data-oe-id]");
+                        this.services.dialog.add(MenuDialog, {
+                            name: menuEl.textContent,
+                            url: menuEl.parentElement.attributes["href"].nodeValue,
+                            save: (name, url) => {
+                                const websiteId = this.services.website.currentWebsite.id;
+                                const data = {
+                                    id: parseInt(menuEl.attributes["data-oe-id"].nodeValue),
+                                    name,
+                                    url,
+                                };
+                                return this.services.orm
+                                    .call("website.menu", "save", [websiteId, { data: [data] }])
+                                    .then(function () {
+                                        menuEl.parentElement.attributes["href"].nodeValue = url;
+                                        menuEl.textContent = name;
+                                    });
```

Comment:
#```suggestion
                                const result = await this.services.orm
                                    .call("website.menu", "save", [websiteId, { data: [data] }])
                                        menuEl.parentElement.attributes["href"].nodeValue = url;
                                        menuEl.textContent = name;
```
I alway prefer to use await 
I just use .then if I can t use await